worldedit-core/src/main/java/com/sk89q/worldedit/util/schematic/Schematic.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util.schematic;
+
+import java.nio.file.Path;
+
+/**
+ * Record representing one Schematic file.
+ */
+public record Schematic(Path path) {
```

Comment:
I haven't had a complete chance to review the PR yet sorry, but it definitely looks great - thanks for putting this together :) 

One part I'm not super sure about is calling the record that represents a schematic path just `Schematic`. Given we might theoretically have data types that represent the data stored in a schematic file in the future, or that users might get confused about what this data actually represents, it might make more sense to give it a name that implies it's purely the path. For example, `SchematicLocator`, `SchematicDescriptor`, `SchematicPath`, etc. I somewhat feel `SchematicPath` might make the most sense, given it feels like a specialisation of the `Path` type for schematics